server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/sql/Db2SqlVisitor.java

```diff
@@ -110,6 +110,16 @@ public List<Node> visitLob_host_variables(Db2SqlParser.Lob_host_variablesContext
         return hostVariableDefinitionNode;
     }
 
+    @Override
+    public List<Node> visitLob_host_variables_arrays(Db2SqlParser.Lob_host_variables_arraysContext ctx) {
+        List<Node> hostVariableDefinitionNode = createHostVariableDefinitionNode(ctx, ctx.dbs_host_var_levels_arrays(), ctx.entry_name());
+        if (ctx.lobWithSize() != null && ctx.lobWithSize().BINARY() != null) {
+            generateVarbinVariables((VariableDefinitionNode) hostVariableDefinitionNode.get(0),
```

Comment:
don't see use case tests for these

Reply:
This would also apply to the `BLOB` variant, not just the `BINARY LARGE OBJECT`, right?

Reply:
One more question to this - what is the origin of the restriction to binary lobs only?
I see this in the DB2 preprocessor output:
```
 04  VAR4                             OCCURS 30.           
   49 VAR4-LENGTH                    PIC S9(9) COMP-5.     
   49 VAR4-DATA                      PIC X(20).            
*   04 VAR4 USAGE IS SQL TYPE IS BLOB (20) OCCURS 30 TIMES.
 04  VAR5                             OCCURS 30.           
   49 VAR5-LENGTH                    PIC S9(9) COMP-5.     
   49 VAR5-DATA                      PIC X(20).            
*   04 VAR5 USAGE IS SQL TYPE IS CLOB (20) OCCURS 30 TIMES.
 04  VAR6                             OCCURS 40.           
   49 VAR6-LENGTH                    PIC S9(9) COMP-5.     
   49 VAR6-DATA                      PIC G(30720) DISPLAY-1
*   04 VAR6 USAGE IS SQL TYPE IS DBCLOB (30 K) OCCURS 40.  
```

Reply:
> One more question to this - what is the origin of the restriction to binary lobs only? I see this in the DB2 preprocessor output:
> 
> ```
>  04  VAR4                             OCCURS 30.           
>    49 VAR4-LENGTH                    PIC S9(9) COMP-5.     
>    49 VAR4-DATA                      PIC X(20).            
> *   04 VAR4 USAGE IS SQL TYPE IS BLOB (20) OCCURS 30 TIMES.
>  04  VAR5                             OCCURS 30.           
>    49 VAR5-LENGTH                    PIC S9(9) COMP-5.     
>    49 VAR5-DATA                      PIC X(20).            
> *   04 VAR5 USAGE IS SQL TYPE IS CLOB (20) OCCURS 30 TIMES.
>  04  VAR6                             OCCURS 40.           
>    49 VAR6-LENGTH                    PIC S9(9) COMP-5.     
>    49 VAR6-DATA                      PIC G(30720) DISPLAY-1
> *   04 VAR6 USAGE IS SQL TYPE IS DBCLOB (30 K) OCCURS 40.  
> ```

Someone checked that on the mainframe. Probably only binary lobs was checked at that time.
I will make changes for CLOB and DBLOB

Reply:
One last details - DBCLOB generates `PIC G(...)`, not `PIC X(...)`.